src/OrchardCore.Modules/OrchardCore.Search/Drivers/SearchSettingsDisplayDriver.cs

```diff
@@ -57,18 +57,21 @@ public override async Task<IDisplayResult> UpdateAsync(SearchSettings section, B
         {
             var user = _httpContextAccessor.HttpContext?.User;
 
-            if (context.GroupId != GroupId || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageSearchSettings))
+            if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageSearchSettings))
             {
                 return null;
             }
 
-            var model = new SearchSettingsViewModel();
-
-            if (await context.Updater.TryUpdateModelAsync(model, Prefix))
+            if (context.GroupId.Equals(GroupId, StringComparison.OrdinalIgnoreCase))
```

Comment:
I would add his check back to line 60 to reduce the indentation.

Reply:
We did the same check after permission check in all the places